### PR TITLE
Lpa 3104 add tmla end point

### PR DESCRIPTION
**Pull request by ramchale**

### CODE REVIEW

File: config/autoload/global.php [hunk 1]
```diff
@@ -163,4 +163,9 @@
             'auto_create' => getenv('OPG_LPA_COMMON_DYNAMODB_AUTO_CREATE') ?: false,
         ],
     ], // session
+
+    'processing-status' => [
+        'track-from-date' => '2015-01-01',
```
Comment (ramchale) [inline config/autoload/global.php]: Should also get the value from environment variables, like the one below

File: module/Application/src/Controller/StatusController.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    /**
+     * @var AuthorizationService AuthorizationService
+     */
+    private $service;
+
+    /**
+     * @var $authorizationService Service
+     */
+    private $authorizationService;
```
Comment (ramchale) [inline module/Application/src/Controller/StatusController.php]: This and $service have the types wrong

File: module/Application/src/Model/DataAccess/Postgres/ApplicationData.php [hunk 3]
```diff
@@ -224,11 +224,13 @@ public function update(Lpa $lpa) : bool
         if (!is_null($inDbLpa)) {
             $inDbLpa = new Lpa($inDbLpa);
 
-            if ($inDbLpa->isLocked()) {
+            $noDataChanged = $lpa->equalsIgnoreMetadata($inDbLpa);
```
Comment (ramchale) [inline module/Application/src/Model/DataAccess/Postgres/ApplicationData.php]: Should add a test for this

File: module/Application/src/Model/Service/ProcessingStatus/Service.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            'Accept'        => 'application/json',
+            'Content-type'  => 'application/json'
+        ];
+
+        return $headers;
+    }
+
+    private function handleResponse(ResponseInterface $response)
+    {
+        $map = ['Pending' => 'Received',
```
Comment (ramchale) [inline module/Application/src/Model/Service/ProcessingStatus/Service.php]: Move this up to a class constant

File: module/Application/src/Model/Service/ProcessingStatus/Service.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            $status = json_decode($response->getBody(), true);
+
+            // TODO - check if found?
+
+            //  If the body isn't an array now then it wasn't JSON before
+            if (!is_array($status)) {
+                throw new ApiProblemException($response, 'Malformed JSON response from server');
+            }
+
+            return $map[$status['status']];
```
Comment (ramchale) [inline module/Application/src/Model/Service/ProcessingStatus/Service.php]: As the TODO says, we need to check $status['found'] and return null if false

File: module/Application/tests/Model/Service/ProcessingStatus/ServiceTest.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+        $this->setUpRequest();
+
+        $result = $this->service->getStatus(12);
+
+        $this->assertEquals('Concluded', $result);
+
+
+    }
+
```
Comment (ramchale) [inline module/Application/tests/Model/Service/ProcessingStatus/ServiceTest.php]: Add test for 200, '{"found": false}' should return null

File: module/Application/tests/Model/Service/ServiceAbstractFactoryTest.php [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    public function testInvoke($service, $dependancies)
+    {
+        foreach($dependancies as $key => $value){
+            $this->container->shouldReceive('get')->withArgs([$key])->once()->andReturn($value);
+        }
+
+        $this->factory->__invoke($this->container, $service);
+    }
+
```
Comment (ramchale) [inline module/Application/tests/Model/Service/ServiceAbstractFactoryTest.php]: Trying something new with a test data provider, does this make sense to people or is it too difficult to work out what is going on?

File: module/Application/src/Controller/StatusController.php (hunk 2 above)
Comment (SeemaMenon) [reply module/Application/src/Controller/StatusController.php]: done

File: module/Application/src/Model/Service/ProcessingStatus/Service.php (hunk 4 above)
Comment (SeemaMenon) [reply module/Application/src/Model/Service/ProcessingStatus/Service.php]: done

File: module/Application/src/Model/Service/ProcessingStatus/Service.php (hunk 5 above)
Comment (SeemaMenon) [reply module/Application/src/Model/Service/ProcessingStatus/Service.php]: done

File: module/Application/tests/Model/Service/ProcessingStatus/ServiceTest.php (hunk 6 above)
Comment (SeemaMenon) [reply module/Application/tests/Model/Service/ProcessingStatus/ServiceTest.php]: done
